source/agora/common/DNS.d

```diff
@@ -845,7 +849,11 @@ public struct ResourceRecord
                 case TYPE.SOA:
                     return this.rdata.soa.serializeFull(CompactMode.No);
                 case TYPE.URI:
-                    return this.rdata.uri.serializeFull(CompactMode.No);
+                    ubyte[] buffer;
+                    buffer ~= this.rdata.uri.priority.serializeFull(CompactMode.No);
+                    buffer ~= this.rdata.uri.weight.serializeFull(CompactMode.No);
+                    buffer ~= this.rdata.uri.target.toString().representation;
```

Comment:
So, this allocates, which we'd like to avoid. Can we avoid it ?
Taking a look at `URL` shows there's an overload that accepts an `OutputRange`:
https://github.com/vibe-d/vibe.d/blob/83753ac1f58e2b8abd26cca1f0730a67d09ace64/inet/vibe/inet/url.d#L381-L382

(Well to be fair [we put it here](https://github.com/vibe-d/vibe.d/commit/0bd793c7ee15483b5e0de8f8ec5a382f9e916119)).

So if we can wrap our `dg` in an `OutputRange`, we're good. Note that the `OutputRange` is a structural concept (like any range), hence there's many ways one can approach this. E.g.
```D
static struct ORWrapper
{
    private SerializeDg dg;
    
    public void put (char c) @trusted { dg((cast(ubyte*) &c)[0 .. 1]); }
    public void put (in char[] s) @safe { dg(s.representation); }
}
```

Reply:
Although... Now this comment doesn't make sense anymore, since the other places allocate too.